src/go/k8s/pkg/resources/certmanager/type_helpers.go

```diff
@@ -199,48 +208,41 @@ func NewClusterCertificates(
 		internalFQDN: fqdn,
 		clusterFQDN:  clusterFQDN,
 		logger:       logger,
-
-		kafkaAPI:          tlsDisabledAPICertificates(),
-		schemaRegistryAPI: tlsDisabledAPICertificates(),
-		adminAPI:          tlsDisabledAPICertificates(),
-		pandaProxyAPI:     tlsDisabledAPICertificates(),
 	}
 	var err error
-	if kafkaListeners := kafkaAPIListeners(cluster); len(kafkaListeners) > 0 {
-		cc.kafkaAPI, err = cc.prepareAPI(ctx, kafkaAPI, RedpandaNodeCert, []string{OperatorClientCert, UserClientCert, AdminClientCert}, kafkaListeners, &keystoreSecret)
-		if err != nil {
-			return nil, fmt.Errorf("kafka api certificates %w", err)
-		}
+	kafkaListeners := kafkaAPIListeners(cluster)
```

Comment:
The listeners list _can_ be nil. I think we should leave the guard on this and consider (in another issue) adding a test to exercise the nil list. wdyt?

Reply:
ok. I will add test.

Reply:
Actually many existing unit tests failed after making the changes. I am not sure what the desired behavior is. If the listeners are mandatory, suggest we make the changes in separate PR.